baybe/campaign.py

```diff
@@ -222,6 +265,72 @@ def add_measurements(
             numerical_measurements_must_be_within_tolerance,
         )
 
+    def _mark_as_measured(
+        self,
+        measurements: pd.DataFrame,
+        numerical_measurements_must_be_within_tolerance: bool,
+    ) -> None:
+        """Mark the given elements of the discrete subspace as measured.
+
+        Args:
+            measurements: A dataframe containing parameter configurations to be
+                marked as measured.
+            numerical_measurements_must_be_within_tolerance: See
+                :func:`baybe.utils.dataframe.fuzzy_row_match`.
+        """
+        idxs_matched = fuzzy_row_match(
+            self.searchspace.discrete.exp_rep,
+            measurements,
+            self.parameters,
+            numerical_measurements_must_be_within_tolerance,
+        )
+        self._searchspace_metadata.loc[idxs_matched, _MEASURED] = True
+
+    def toggle_discrete_candidates(
+        self,
+        filter: pd.DataFrame,
+        exclude: bool,
+        anti: bool = False,
+        dry_run: bool = False,
+    ) -> pd.DataFrame:
+        """Un-/exclude certain discrete points from the candidate set.
+
+        Args:
+            filter: A dataframe specifying the filtering mechanism to determine the
+                candidates subset to be un-/excluded. The subset is determined via a
+                join (see ``anti`` argument for details) with the discrete space.
+            exclude: If ``True``, the specified candidates are excluded.
+                If ``False``, the candidates are considered for recommendation.
+            anti: If ``False``, the filter determines the points to be affected (i.e.
```

Comment:
I also saw that, but it's really counterintuitive / impossible to mention an "inversion" when you haven't described the base case first. So probably one could completely rewrite the docstring somehow, but I don't see a way that doesn't make the logic harder to understand. Also, it would then come with the problem that we describe the non-default first. 

But still, feel free to suggest a better alternative and we can consider it 👍🏼

Reply:
I think the general description is quite good, and I understand your points specifically regarding the "Explain default first" - didn't notice that. Hence let me resolve